### PR TITLE
Branch stratio spark development

### PR DESCRIPTION
**Pull request by jlopezmalla**
## What changes were proposed in this pull request?
calico needs two networks one for the Driver and another for the Executrors so we need to change this configuration:

`spark.mesos.docker.mesos.network.name`

for:
````
spark.mesos.driver.docker.mesos.network.name
spark.mesos.executor.docker.mesos.network.name
````
Besides we need two images one for the Spark HistoryServer and another for the Stratio Spark Dispatcher.

## How was this patch tested?

These patch hasn't been tested.

### CODE REVIEW

File: DockerfileHistory [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    && yum -y upgrade \
+    && yum -y install java-1.8.0-openjdk curl wget\
+    && yum -y clean all
+
+RUN wget https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64 \
+    && mv jq-linux64 /usr/local/bin/jq \
+    && chmod 755 /usr/local/bin/jq\
+    && ln -s /usr/local/bin/jq /usr/bin/
+
+RUN mkdir -p /var/sds/spark/
```
Comment (mpenate) [inline DockerfileHistory]: separate run blocks ends up in more layers

File: DockerfileHistory [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+RUN wget -O /usr/local/bin/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_amd64 && chmod +x /usr/local/bin/dumb-init
+
+COPY dist /opt/sds/spark
+
+COPY docker/historyServer/entrypoint.sh /
```
Comment (mpenate) [inline DockerfileHistory]: copy all files from using only docker/historyServer is an option?

File: Jenkinsfile [hunk 3]
```diff
@@ -3,17 +3,18 @@
...
     DEVTIMEOUT = 40
     RELEASETIMEOUT = 40
-    PKGMODULESNAMES = ['spark-stratio-2_11-R2']
+    PKGMODULESNAMES = ['spark-stratio-2_11-r2']
 
     DEV = { config ->
 
         doPackage(config)
-	doDocker(config)
+	    doDocker(conf: config, dockerfile:"DockerfileDispatcher")
```
Comment (mpenate) [inline Jenkinsfile]: indentation seems to be wrong

File: docker/dispatcher/init.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -e
+set -x
+
+export DISPATCHER_PORT=7077
+export DISPATCHER_UI_PORT=7076
+
+# determine scheme and derive WEB
+SCHEME=http
```
Comment (mpenate) [inline docker/dispatcher/init.sh]: why do you use these vars? maybe:

if [[ "${SPARK_SSL_ENABLED}" == true ]]; then
	SCHEME=https
else
        SCHEME=http
fi

File: docker/historyServer/commons.sh [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    lw_realm=$(echo $realm | tr '[:upper:]' '[:lower:]')
+
+    cat << EOF > /tmp/krb5.conf.tmp
+[libdefaults]
+ default_ccache_name = /tmp/%{uid}/krb5cc_%{uid}
+ default_realm = ${realm}
+ dns_lookup_realm = false
+ dns_lookup_kdc = false
+ udp_preference_limit = 1
+ renew_lifetime = 7d
```
Comment (mpenate) [inline docker/historyServer/commons.sh]: this is not inside a variable, is there any reason? shall we let it be as a var?

File: docker/historyServer/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+   mv "/tmp/core-site.xml.tmp" "${HADOOP_CONF_DIR}/core-site.xml"
+
+   # Needed variables to generate a hdfs-site.xml.
+
+   generate_hdfs_site "${HDFS_DFS_PERMISSIONS_ENABLED}" "${HDFS_DFS_BLOCK_ACCESS_TOKEN_ENABLE}" "${HDFS_DFS_HTTP_POLICY}" "${HDFS_DFS_HTTPS_PORT}"
+   mv "/tmp/hdfs-site.xml.tmp" "${HADOOP_CONF_DIR}/hdfs-site.xml"
+
+   SPARK_HISTORY_OPTS="-Dspark.history.fs.logDirectory=hdfs://$HDFS_FS_DEFAULTFS/${HISTORY_SERVER_LOG_DIR} ${SPARK_HISTORY_OPTS}"
+
+   echo $SPARK_HISTORY_OPTS
```
Comment (mpenate) [inline docker/historyServer/entrypoint.sh]: is this echo needed? or it just had debug purpose?

File: docker/historyServer/kms_utils.sh [hunk 7]
```diff
@@ -0,0 +1,438 @@
+#!/bin/bash
```
Comment (mpenate) [inline docker/historyServer/kms_utils.sh]: @mlorenzo-stratio please, decide if this is a valid version with all the new checks you added

File: docker/dispatcher/init.sh (hunk 4 above)
Comment (jlopezmalla) [inline docker/dispatcher/init.sh]: Legacy code, when we will have a better understanding of its purpose we could delete it

**Comment by jlopezmalla** [comment]
Lgtm And merging

**Pull request merged**